daemon/cluster/swarm.go

```diff
@@ -25,19 +25,20 @@ import (
 func (c *Cluster) Init(req types.InitRequest) (string, error) {
 	c.controlMutex.Lock()
 	defer c.controlMutex.Unlock()
-	c.mu.Lock()
```

Comment:
given we remove the big mu.Lock section here, which is good, but then we should acquire `nr:=c.nr` under lock and then use `nr` throughout this function. otherwise now we freely use `c.nr` with the potential risk it gets modified in between.

Or is there an implicit assurance `c.nr` cannot be modified because we hold the `c.controlMutex.Lock()` ? 
If that is true, then it should have been true in Cleanup() as well.